scripts/analyze.py

```
@@ -0,0 +1,462 @@
+            freq[e] += 1
+
+    total_unique = len(freq)
+    total_with_freq = sum(freq.values())
+
+    if rank_by_frequency:
+        sorted_entries = sorted(freq.items(), key=lambda x: (-x[1], x[0]))
+        merged = [entry for entry, count in sorted_entries]
+        write_lines = [f"{entry}\n" for entry in merged]
+    else:
+        merged = list(freq.keys())
+        write_lines = sorted(merged + ['\n'])
```

Ruff (0.15.17)

[warning] Consider `[*merged, '\n']` instead of concatenation

Replace with `[*merged, '\n']`

(RUF005)